src/form.js

```diff
@@ -103,15 +103,18 @@ export default function higherform(fieldSpec, formSpec) {
                 this.getData = this.getData.bind(this);
             }
 
-            componentWillReceiveProps(nextProps) {
-                let fieldsChanged = this._configureFields(nextProps);
-                let newFormData = finalFormSpec.nextPropsToForm(this.props, nextProps);
-                if (!!newFormData) {
-                    this.setState(filterFormData(newFormData, this.fields));
-                } else if (fieldsChanged) {
-                    // if the fields updated, we need to ensure the state that's
-                    // tracked by this class is up to date with the fields
-                    this.setState(filterFormData(this.state, this.fields));
+            componentDidUpdate(prevProps) {
+                if (prevProps != this.props) {
```

Comment:
I don't think you need to check equality here, won't the props always be a new object and fail this check?